### DISCUSSION TITLE
Required property with sequence

### DISCUSSION THREAD

**Discussion by tchak** [geldata/gel#3493] [Q&A] [answered]
I have the following schema:

```
module default {
  scalar type ProjectNumber extending sequence;

  type Project {
    required property number -> ProjectNumber {
      constraint exclusive;
    }
    required property name -> str;
  }
}
```

I expect `number` to be autofilled. But when I use this schema from TypeScript driver type signature refuses to let me create a Project without specifying a `number` property. I tried the following schema but it failed on migration:

```
module default {
  scalar type ProjectNumber extending sequence;

  type Project {
    required property number -> ProjectNumber {
      constraint exclusive;
      default := sequence_next(introspect ProjectNumber);
    }
    required property name -> str;
  }
}
```

Is this is a bug or intended behaviour ?

**Comment by elprans** [comment, marked as answer]
It's a bug

**Comment by colinhacks** [comment]
@tchak This is fixed in v0.19.3 

https://github.com/edgedb/edgedb-js/pull/253
